BlueDBOnDisk/src/main/java/io/bluedb/disk/backup/BackupTask.java

```diff
@@ -0,0 +1,82 @@
+package io.bluedb.disk.backup;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import io.bluedb.api.exceptions.BlueDbException;
+import io.bluedb.disk.BlueDbOnDisk;
+import io.bluedb.disk.collection.BlueCollectionOnDisk;
+import io.bluedb.disk.file.FileManager;
+import io.bluedb.disk.lock.BlueReadLock;
+import io.bluedb.disk.recovery.RecoveryManager;
+import io.bluedb.disk.segment.Range;
+import io.bluedb.disk.segment.Segment;
+
+public class BackupTask {
+	
+	private final BlueDbOnDisk db;
+	private final Path backupPath;
+	
+	public BackupTask(BlueDbOnDisk db, Path backupPath) {
+		this.db = db;
+		this.backupPath = backupPath;
+	}
+
+	public void backup(List<BlueCollectionOnDisk<?>> collectionsToBackup) throws BlueDbException {// TODO filesystem to decide which to back up?  Or take arguments?>
+		long backupStartTime = System.currentTimeMillis();
+		for (BlueCollectionOnDisk<?> collection: collectionsToBackup) {
+			copyMetaData(collection, backupPath);
+			copyDataFolders(collection, backupPath);
+		}
+		long backupEndTime = System.currentTimeMillis();
+		for (BlueCollectionOnDisk<?> collection: collectionsToBackup) {
+			copyChanges(collection, backupStartTime, backupEndTime);
+		}
+	}
+
+	private void copyMetaData(BlueCollectionOnDisk<?> collection, Path backupPath2) throws BlueDbException {

```

Comment:
Looks like `backupPath2` is never used in these methods. We should remove it since it is available as a final field variable.

Reply:
removed